tests/connection/broker.cpp

```diff
@@ -0,0 +1,71 @@
+
+/* Copyright (c) 2014-2015, Human Brain Project
+ *                          Stefan.Eilemann@epfl.ch
+ */
+
+#define BOOST_TEST_MODULE zeq_connection_broker
+
+#include "../broker.h"
+#include <zeq/connection/broker.h>
+
+#include <lunchbox/sleep.h>
+#include <lunchbox/thread.h>
+#include <lunchbox/servus.h>
+#include <boost/bind.hpp>
+
+using boost::lexical_cast;
+
+const unsigned short port = (lunchbox::RNG().get<uint16_t>() % 60000) + 1024;
+
+class Service : public lunchbox::Thread
+{
+public:
+    Service() : received( false ) {}
+
+    void run() final
+    {
+        zeq::Subscriber subscriber( lunchbox::URI( "foo://127.0.0.1:" +
+                                          lexical_cast< std::string >( port )));
+        BOOST_CHECK( subscriber.registerHandler( zeq::vocabulary::EVENT_ECHO,
+                                        boost::bind( &test::onEchoEvent, _1 )));
+
+        // Using the connection broker in place of zeroconf
+        std::string address = std::string( "127.0.0.1:" ) +
```

Comment:
All the IPs being 127.0.0.1 doesn't help to distinguish which is A and which B. Can you use a different variable for each participant? then this code can be used as an example in the documentation.